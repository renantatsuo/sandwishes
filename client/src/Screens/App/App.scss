.app--card {
  width: 1040px;
  height: 850px;
  position: relative;

  background: #ffffff;
  border: 1px solid #e5e5e5;
  box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.05);
  border-radius: 3px;
  padding: 50px;

  .app--header {
    width: 100%;
    margin-bottom: 70px;

    img {
      max-width: 250px;
    }
  }

  .app--content {
    display: flex;
  }

  &::after {
    content: 'SandWishes 2019';
    font-weight: 300;
    font-size: 12px;
    bottom: 30px;
    position: absolute;
    width: 200px;
    left: 50%;
    margin-left: -100px;
    text-align: center;
  }
}
