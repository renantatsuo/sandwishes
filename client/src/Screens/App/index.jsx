import React, { Component } from 'react'

import './App.scss'
import Logo from '../../Assets/logo.svg'
import Burger from '../../Assets/burger.svg'
import MenuItem from '../../Components/MenuItem/index'
import {
  calculatePrice,
  fetchSandwiches,
  fetchIngredients
} from '../../Helpers'

class App extends Component {
  state = {
    sandwiches: []
  }

  componentWillMount() {
    fetchSandwiches().then((sandwiches) => {
      fetchIngredients().then((ingredients) => {
        this.setState({
          sandwiches: sandwiches.map((sandwich) => ({
            ...sandwich,
            price: calculatePrice(sandwich.recipe, ingredients)
          }))
        })
      })
    })
  }

  mapSandwiches = (sandwiches) => {
    return sandwiches.map((sandwich) => (
      <MenuItem
        key={sandwich.id}
        photo={Burger}
        title={sandwich.name}
        desc={sandwich.description}
        price={sandwich.price}
      />
    ))
  }

  render() {
    const { sandwiches } = this.state

    return (
      <div className="app__card">
        <div className="app__header">
          <img src={Logo} alt="Sand Wishes Logo" />
        </div>
        <div className="app__content">
          <div className="app__diy">
            <button type="button" className="app__diy__button">
              Monte você mesmo
            </button>
          </div>
          <div className="app__menu">
            <h1>Escolher no cardápio</h1>
            {this.mapSandwiches(sandwiches)}
          </div>
        </div>
      </div>
    )
  }
}

export default App
