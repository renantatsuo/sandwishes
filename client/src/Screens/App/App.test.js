import React from 'react'
import { render } from '@testing-library/react'

import App from '.'

describe('Test App screen', () => {
  it('renders without crashing', () => {
    const { container } = render(<App />)

    expect(container.querySelectorAll('.app__header')).toHaveLength(1)
    expect(container).toMatchSnapshot()
  })
})
