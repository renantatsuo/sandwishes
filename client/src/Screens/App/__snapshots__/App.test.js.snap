// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App screen renders without crashing 1`] = `ShallowWrapper {}`;
